core/array/pack/buffer_spec.rb

```diff
@@ -0,0 +1,52 @@
+# encoding: ascii-8bit
+
+require File.expand_path('../../../../spec_helper', __FILE__)
+
+ruby_version_is '2.4' do
+  describe "Aray#pack with `buffer` option" do
+    it "returns specified buffer" do
+      n = [ 65, 66, 67 ]
+      buffer = " "*3
+      result = n.pack("ccc", buffer: buffer)      #=> "ABC"
+      result.object_id.should == buffer.object_id
```

Comment:
There is a built-in matcher to compare identity: equal which calls `equal?`.
`result.should equal buffer`

Reply:
@eregon done